### ISSUE TITLE
JSDoc?

### ISSUE THREAD

**Issue by winniehell** [GitbookIO/gitbook#559]
What do you think of spreading [JSDoc tags](https://github.com/jsdoc3/jsdoc) over the code? Do you know of a better tool for JavaScript API documentation?

**Comment by bebraw** [comment]
If you are going to do that for type annotations, might as well try [flow](http://flowtype.org/) and such.

**Comment by KevinAst** [comment]
Just noticed your question on JSDoc.  I am a strong proponent of this.

I recently experimented with integrating JSDoc (for my API) with GitBook (for my overall Guide).

Here is my docs site: https://astx-redux-util.js.org/

- I am promoting a very polished set of docs using Gitbook, while still relying on my embedded JSDoc comments for the API.

- The **secret in accomplishing this integration** is to use the [jsdoc-to-markdown](https://www.npmjs.com/package/jsdoc-to-markdown) package to generate the low-level API as Markdown (from the JSDoc comments in your code).  Once this is accomplished, it can be easily integrated into the overall GitBook Markdown structure.

- I have created a **comprehensive [gist](https://gist.github.com/KevinAst/7e12648245ff2a8e9c1557135014b933) that details how I accomplished this**.

I would encourage you to explore this technique further.

Thoughts?

</Kevin>
